src/ZODB/tests/StorageTestBase.py

```diff
@@ -113,18 +113,24 @@ def handle_all_serials(oid, *args):
 
     The original interface just returned the serialno for the
     object.
+
+    The updated multi-commit API returns nothing from store(), and
+    returns a sequence of resolved oids from tpc_vote.
     """
     d = {}
     for arg in args:
         if isinstance(arg, bytes):
             d[oid] = arg
-        elif arg is None:
-            pass
-        else:
-            for oid, serial in arg:
-                if not isinstance(serial, bytes):
-                    raise serial # error from ZEO server
-                d[oid] = serial
+        elif arg:
+            for t in arg:
+                if isinstance(t, bytes):
+                    # This will be the tid returned by tpc_finish.
```

Comment:
> On Jul 12, 2016, at 11:14, Jim Fulton <notifications@github.com> wrote:
> 
> The right thing to do would be to look at the call points and decide WTF.
> 
> Given that this is the 4 branch, I guess I'd be willing to accept this if the tests pass and if we leave an XXX comment here saying that the this should get another review after reviewing the call points.
> 

Happily the tests do all pass. 

I just reviewed the callers for #87 (in the process of removing them as requested), and they all do exactly the same thing as the test here did:

        r1 = self._storage.store(...)
        r2 = self._storage.tpc_vote(txn)
        serial = self._storage.tpc_finish(txn)
        newrevid = handle_serials(oid, r1, r2)
        if newrevid is None and serial is not None:
            newrevid = serial

For whatever that's worth...